core/services/chainlink/application.go

```diff
@@ -192,13 +197,79 @@ type ApplicationOpts struct {
 	NewOracleFactoryFn         standardcapabilities.NewOracleFactoryFn
 }
 
+type ApplicationHeartbeat struct {
+	commonservices.Service
+	eng *commonservices.Engine
+
+	beat time.Duration
+	lggr logger.Logger
+}
+
+func NewApplicationHeartbeat(lggr logger.Logger) ApplicationHeartbeat {
+	h := ApplicationHeartbeat{
+		beat: ApplicationHeartbeatSeconds * time.Second,
+		lggr: lggr,
+	}
+	h.Service, h.eng = commonservices.Config{
+		Name:  "NodeHeartbeat",
+		Start: h.start,
+		Close: h.close,
+	}.NewServiceEngine(lggr)
+	return h
+}
+
+func (h *ApplicationHeartbeat) start(_ context.Context) error {
+	// Setup beholder resources
+	gauge, err := beholder.GetMeter().Int64Gauge("heartbeat")
+	if err != nil {
+		return err
+	}
+	count, err := beholder.GetMeter().Int64Gauge("heartbeat_count")
+	if err != nil {
+		return err
+	}
+
+	cme := custmsg.NewLabeler()
+
+	// Define tick functions
+	beatFn := func(engCtx context.Context) {
+		// TODO allow override of tracer provider into engine for beholder
+		_, innerSpan := beholder.GetTracer().Start(engCtx, "heartbeat.beat")
+		defer innerSpan.End()
+
+		gauge.Record(engCtx, 1)
+		count.Record(engCtx, 1)
+
+		err = cme.Emit(engCtx, "heartbeat")
+		if err != nil {
+			h.lggr.Errorw("heartbeat emit failed", "err", err)
+		}
+	}
+
+	// consistent tick period
+	constantTickFn := func() time.Duration {
+		return h.beat
+	}
+
+	h.eng.GoTick(timeutil.NewTicker(constantTickFn), beatFn)
+	return nil
+}
```

Comment:
A method might be simpler here:
```suggestion
	h.eng.GoTick(timeutil.NewTicker(h.getBeat), beatFn)
	return nil
}

func (h *ApplicationHeartbeat) getBeat() time.Duration { return h.beat }
```